src/wavm/WAVMWasmModule.cpp

```diff
@@ -851,29 +817,13 @@ uint32_t WAVMWasmModule::addFunctionToTable(Runtime::Object* exportedFunc)
     return prevIdx;
 }
 
-/**
- * Executes the given function call
- */
-bool WAVMWasmModule::execute(faabric::Message& msg, bool forceNoop)
+int32_t WAVMWasmModule::executeFunction(faabric::Message& msg)
 {
-    const std::shared_ptr<spdlog::logger>& logger = faabric::util::getLogger();
-
     if (!_isBound) {
-        throw std::runtime_error(
-          "WasmModule must be bound before executing function");
-    } else {
-        if (boundUser != msg.user() || boundFunction != msg.function()) {
-            const std::string funcStr = faabric::util::funcToString(msg, true);
-            logger->error("Cannot execute {} on module bound to {}/{}",
-                          funcStr,
-                          boundUser,
-                          boundFunction);
-            throw std::runtime_error(
-              "Cannot execute function on module bound to another");
-        }
+        throw std::runtime_error("Module must be bound before executing");
```

Comment:
Maybe another use case for `assertm()`?

Reply:
Hmm it's a good question. I think `assertm` is probably never preferable to an `if` and an exception, other than the fact that it saves some lines. I do like the fact that `assertm` can be done in one line, hence doesn't clutter the code, but because it's removed in non-debug builds it risks missing errors in deployments. It would be nice to save some lines on checks like this though, so perhaps it should be replaced with an `assertm`-like macro that doesn't get removed.

I would say we should only use `assert` in places where we're checking things that, in theory, will never go wrong, e.g. checking arithmetic in a loop, testing that something isn't a nullptr in some internal accessor method. I guess I think of them as extensions of tests, where if you can look at it and say "ah that's obviously wrong". However, it's not clear-cut and perhaps we should create something like `assertm` but not actually using `assert`